.github/workflows/main.yaml

```diff
@@ -1,977 +1,95 @@
-name: "Build contracts"
```

Comment:
Here looks like the main task and the related tests correctly running on the developers fork: https://github.com/bojinovic/filecoin-solidity-library/actions/runs/9063375701/job/24899183910

We need to confirm that this works on this repo after we merge it too